spec/models/miq_report/seeding_spec.rb

```diff
@@ -1,5 +1,95 @@
+require 'fileutils'
+
 describe MiqReport do
-  context "Seeding" do
+  describe "::Seeding" do
     include_examples(".seed called multiple times", 143)
+
+    let(:connection) { ApplicationRecord.connection }
+
+    describe ".seed" do
+      let(:report_dir)        { described_class::REPORT_DIR.join("1_Testing Group") }
+      let(:source_report_yml) { Pathname.new(__dir__).join("data/022_Testing Report.yaml") }
+      let(:report_yml)        { report_dir.join("022_Testing Report.yaml") }
+
+      let(:compare_dir)        { described_class::COMPARE_DIR }
+      let(:source_compare_yml) { Pathname.new(__dir__).join("data/testing.yaml") }
+      let(:compare_yml)        { compare_dir.join("testing.yaml") }
+
+      after do
+        FileUtils.rm_f(report_yml)
+        FileUtils.rm_f(compare_yml)
+      end
+
+      # This test is intentionally long winded instead of breaking it up into
+      # multiple tests per concern because of how long a full seed takes.
+      # Breaking it into individual tests would increase runtime a lot.
+      it "creates, updates, and changes records" do
+        described_class.seed
+
+        expect(MiqReport.where(:name => "Testing Report Name")).to_not exist
+        expect(MiqReport.where(:name => "Testing Compare Name")).to_not exist
+
+        # Add new reports
+        FileUtils.mkdir_p(report_dir)
+        FileUtils.cp(source_report_yml, report_dir)
+        FileUtils.mkdir_p(compare_dir)
+        FileUtils.cp(source_compare_yml, compare_dir)
```

Comment:
could you just point directly to the `report_dir` / `compare_dir`?

Reply:
I'm not sure what you're asking, but perhaps explaining this will answer your question?

The on-disk structure of the product/report and product/compare dirs is actually important to seeding.  That is, you can't tell seeding to point to alternate dirs in addition.  So, what this is doing is

a) create a target directory in the existing product/report and product/compare structure.
b) copy some pre-made reports from spec data into that structure
c) Later in the after callback, we delete the created file and directory.

Reply:
Copying the files into the normal seeding directory can cause test contamination between tests.  For example, the `include_examples(".seed called multiple times", 143)` could suddenly fail if run in parallel with this test as we could have 145 rows.

We probably need to create a tempdir and copy files into it.  We'll need to worry about the mtime of the copy since your changes here care about the mtime so the copy could screw that up.  We'll then need to remove the files after.  Sorry 😢

Reply:
The other nice benefit of using a tempdir is we don't need to seed all the normal default reports in the test, since we can hand pick a few to do along with your test samples.

Reply:
Update the tests to create their own isolated temp directories